app/data/sendMessageToEmployee.server.ts

```diff
@@ -0,0 +1,91 @@
+import { User } from "@clerk/clerk-sdk-node";
+import getMysql from "./mysql.server";
+import { VellumClient } from "vellum-ai";
+import { WorkflowRequestInputRequest, ChatMessageRequest } from "vellum-ai/api";
+
+const vellum = new VellumClient({
+  apiKey: process.env.VELLUM_API_KEY ?? "",
+});
+
+const sendMessageToEmployee = async ({
+  requestId,
+  message,
+  user,
+}: {
+  requestId: string;
+  message: string;
+  user: Pick<User, "firstName" | "lastName">;
+}) => {
+  await getMysql(requestId);
+  const chatHistory: ChatMessageRequest[] = [
+    {
+      role: "USER",
+      text: message,
+    },
+  ];
+  const inputs: WorkflowRequestInputRequest[] = [
+    {
+      type: "CHAT_HISTORY",
+      name: "$chat_history",
+      value: chatHistory,
+    },
+    {
+      type: "STRING",
+      name: "user",
+      value: `${user.firstName} ${user.lastName}`,
+    },
+  ];
+  const response = await vellum.executeWorkflow({
+    workflowDeploymentName: "chief-of-staff",
+    inputs,
+  });
+  if (response.data.state === "FULFILLED") {
+    const employeeOutput = response.data.outputs.find(
+      (o) => o.name === "employee"
+    );
+    const resultsOutput = response.data.outputs.find(
+      (o) => o.name === "results"
+    );
+    const plainOutput = response.data.outputs.find((o) => o.name === "plain");
+    if (plainOutput) {
+      return {
+        response: plainOutput.value,
+      };
+    }
+    const functionCall = JSON.parse(employeeOutput?.value as string);
+    const functionName = functionCall.function_call.name;
+    chatHistory.push(
+      {
+        role: "ASSISTANT",
+        text: JSON.stringify({ function_call: functionCall }),
+      },
```

Comment:
@mdroidian to answer your question, check this out. The model needs the full history so we're grabbing it to make sure we feed it back for the next step of the convo.

The devX/UX here is horrendous (comes directly from OpenAI), and something I personally will be improving hopefully over the next 3 months.